dialects/hir/src/builders/function.rs

```diff
@@ -1,36 +1,38 @@
 use midenc_hir2::{
-    dialects::builtin::*, AsCallableSymbolRef, Block, BlockRef, Builder, Immediate, Op, OpBuilder,
-    Overflow, Region, RegionRef, Report, SourceSpan, Type, UnsafeIntrusiveEntityRef, Usable,
-    ValueRef,
+    dialects::builtin::*, AsCallableSymbolRef, Block, BlockRef, Builder, Immediate, Listener,
+    OpBuilder, Overflow, Region, RegionRef, Report, SourceSpan, Type, UnsafeIntrusiveEntityRef,
+    Usable, ValueRef,
 };
 
 use crate::*;
 
-pub struct FunctionBuilder<'f> {
+pub struct FunctionBuilder<'f, L: Listener> {
```

Comment:
I'm not sure this is actually needed anymore, no? It looks like the `FunctionBuilder` in `midenc_dialect_hir` already provides equivalent functionality (and more complete, i.e. most instructions are already supported there). If we need this custom one, maybe it would be better to adapt the `midenc_dialect_hir` version to be more generic, than to derive an almost identical one in this crate.

I haven't finished reviewing this PR yet though, so I may be missing the reason why we've kept this around for now.

Reply:
Wait, I'm dumb, ignore this 😅